src/main/java/org/jabref/gui/desktop/os/Linux.java

```diff
@@ -89,41 +90,61 @@ public void openFileWithApplication(String filePath, String application) throws
     public void openFolderAndSelectFile(Path filePath) throws IOException {
         String desktopSession = System.getenv("DESKTOP_SESSION");
 
-        String cmd = "xdg-open " + filePath.toAbsolutePath().getParent().toString(); // default command
+        String cmd = "xdg-open"; // default command
 
         if (desktopSession != null) {
             desktopSession = desktopSession.toLowerCase(Locale.ROOT);
             if (desktopSession.contains("gnome")) {
```

Comment:
Does not happen often, but often enough: people may run a certain desktop environment, but replaced the default file manager. This would break then.
Also, we cannot check for all possible DEs and may fail to open a file in a nice way even though we know the parameters for the file manager because we don't know the DE. I for example run qtile and have dolphin installed. This check would not cover that scenario.
Instead, could we just try all the file-managers that we know one by one and only stop when one does not return an error?

Reply:
Runtime.exec is offically deprecated and they recommend using ProcessBuilder.
The only thing we can do and check if the error stream is not empty and then try the next one...

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/ProcessBuilder.html